### PR TITLE
[Merged by Bors] - Pack attestations into blocks in parallel

### PR DESCRIPTION
**Pull request by michaelsproul**
## Proposed Changes

Use two instances of max cover when packing attestations into blocks: one for the previous epoch, and one for the current epoch. This reduces the amount of computation done by roughly half due to the `O(n^2)` running time of max cover (`2 * (n/2)^2 = n^2/2`). This should help alleviate some load on block proposal, particularly on Prater.

### CODE REVIEW

File: beacon_node/operation_pool/src/lib.rs [hunk 1]
```diff
@@ -96,6 +97,43 @@ impl<T: EthSpec> OperationPool<T> {
...
+                verify_attestation_for_block_inclusion(
+                    state,
+                    attestation,
+                    VerifySignatures::False,
+                    spec,
+                )
+                .is_ok()
+            })
+            .filter(validity_filter)
+            .flat_map(move |att| AttMaxCover::new(att, state, total_active_balance, spec))
```
Comment (paulhauner) [inline beacon_node/operation_pool/src/lib.rs]: Thoughts about using `filter_map` instead? Just from a readability standpoint.

File: beacon_node/operation_pool/src/lib.rs [hunk 2]
```diff
@@ -96,6 +97,43 @@ impl<T: EthSpec> OperationPool<T> {
...
+            state.genesis_validators_root,
+            spec,
+        );
+        all_attestations
+            .iter()
+            .filter(move |(key, _)| key.domain_bytes_match(&domain_bytes))
+            .flat_map(|(_, attestations)| attestations)
+            .filter(move |attestation| {
+                if attestation.data.target.epoch != epoch {
+                    return false;
```
Comment (paulhauner) [inline beacon_node/operation_pool/src/lib.rs]: Out of interest, any reason this isn't it's own `filter` statement?

**Review by paulhauner — COMMENTED** [review]
Looks good! I just had a question and a minor comment :pray:

File: beacon_node/operation_pool/src/lib.rs (hunk 2 above)
Comment (michaelsproul) [reply beacon_node/operation_pool/src/lib.rs]: No good reason! Just haste :sweat_smile:

File: beacon_node/operation_pool/src/lib.rs (hunk 1 above)
Comment (michaelsproul) [reply beacon_node/operation_pool/src/lib.rs]: Good call

**Comment by michaelsproul** [comment]
Thanks! Review comments addressed :rocket:

**Review by paulhauner — APPROVED** [review]
Bors it!

**Pull request closed without merging**